fenix/app/src/main/java/org/mozilla/fenix/FenixOverlay.kt

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.SnackbarHostState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.datastore.debugDrawerEnabled
+import org.mozilla.fenix.datastore.debugSettings
+import org.mozilla.fenix.debugdrawer.ui.DebugDrawerFAB
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Overlay for presenting Fenix-wide content.
+ */
+@Composable
+fun FenixOverlay() {
+    val debugDataStore = LocalContext.current.debugSettings
+    val debugDrawerFabEnabled by debugDataStore.debugDrawerEnabled.collectAsState(initial = false)
+
+    FenixOverlayContent(
+        debugDrawerFabEnabled = debugDrawerFabEnabled,
+    )
+}
+
+@Composable
+private fun FenixOverlayContent(
+    debugDrawerFabEnabled: Boolean,
+) {
+    val snackbarState = remember { SnackbarHostState() }
+    val scope = rememberCoroutineScope()
+
+    Box(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        DebugDrawerFAB(
```

Comment:
nit: Not sure about the positioning of the button, but appreciate this may change or is desirable in this use case

Reply:
For 0.1, I'm thinking center of the screen. A follow-up once the drawer is stable is to make the FAB draggable, so it can be repositioned anywhere on the screen!

Reply:
Maybe center-left to start? Think that would avoid the more common gesture-based back nav, and may leave more of the screen generally interact-able

Reply:
so like center along the left edge of the screen? or biased like 50 just left of center?

Reply:
left edge would be my suggestion. i'm mostly just worried that repositioning it might be hard and so punted on, and center screen seems like important real-estate